didcomm_messaging/quickstart.py

```diff
@@ -0,0 +1,393 @@
+    # DID as their endpoint, in which case, that DID will also be unwrapped
+    # into it's own DIDDocument.
+    #
+    # At present, the PrefixResolver is used to determine which library should
+    # be used to convert a DID into a DIDDocument.
+    if enable_compatibility_prefix:
+        resolver = CompatibilityPrefixResolver(
+            {"did:peer:2": Peer2(), "did:peer:4": Peer4()}
+        )
+    else:
+        resolver = PrefixResolver({"did:peer:2": Peer2(), "did:peer:4": Peer4()})
+
+    # The Packaging Service is where a lot of the magic happens. Similar to a
+    # shipping box, the PackagingService will "pack" and "unpack" an encrypted
+    # message. When packing a message, the PackagingService will encrypt the
+    # message to a single target, however. If the message needs to be forwarded
+    # (because the recipient is behind a relay), then those messages will need
+    # to be handled by the RoutingService.
+    packer = PackagingService(resolver, crypto, secrets)
+
+    # The RoutingService handles the routing of messages through relays. When a
+    # message needs to be forwarded, the RoutingService will handle wrapping
+    # each encrypted message within a forward message. The built-in
+    # RoutingService allows for multiple levels of relays, so you don't need to
+    # worry about how a message is routed to the recipient.
+    router = RoutingService(packaging=packer, resolver=resolver)
+
+    # Once everything is setup, we need to store the DID Secrets within the
+    # SecretsService. We do this by taking the secrets that were passed in,
+    # converting them to AskarSecretKey objects (since we are using the
+    # AskarCryptoService), then we associate them with the public key lookup
+    # value that will be sent to us.
+    verkey, xkey = did_secrets
+    await secrets.add_secret(AskarSecretKey(verkey, f"{did}#key-1"))
+    await secrets.add_secret(AskarSecretKey(xkey, f"{did}#key-2"))
+
+    # These can be removed once all keys being sent/received are in #key-N
+    # format. They basically do the same as above, except with the same public
+    # keys that are returned from the CompatibilityPrefixResolver.
+    doc = await resolver.resolve_and_parse(did)
+    await secrets.add_secret(AskarSecretKey(verkey, doc.authentication[0]))
+    await secrets.add_secret(AskarSecretKey(xkey, doc.key_agreement[0]))
+
+    # Finally, we put it all together in the DIDCommMessaging class. The
+    # DIDCommMessaging handles the orchestration of each individual service,
+    # ensuring that messages get packed and delivered via a simple and straight
+    # forward interface.
+    DMP = DIDCommMessaging(
+        crypto=crypto,
+        secrets=secrets,
+        resolver=resolver,
+        packaging=packer,
+        routing=router,
+    )
+
+    return DMP
+
+
+async def send_http_message(
+    dmp: DIDCommMessaging, my_did: DID, message: Message, target: DID
+):
```

Comment:
I think the method name and signature (return type hint) should indicate that this message can return a reply.